src/Sparrow/ObjectPool.cs

```diff
@@ -103,7 +103,75 @@ private struct CacheAwareElement
             private T Value4;
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            public bool TryClaim(ref CacheAwareElement bucket, out T item)
+            public static void Clear<TEvictionStrategy>(ref CacheAwareElement bucket, ref TEvictionStrategy policy)
+                where TEvictionStrategy : struct, IEvictionStrategy<T>
+            {
+                // Note that the initial read is optimistically not synchronized. That is intentional. 
+                // We will interlock only when we have a candidate. in a worst case we may miss some
+                // recently returned objects. Not a big deal.
```

Comment:
We have the finalizer for cleanup, but that makes things much more complex and delay cleanup